File: c++/src/rib.cpp
/*
 *  rib.cpp
 *  Dragon3D
 *
 *  Created by Olivier Van Acker on 13/06/2010.
 *  Copyright 2010 Transformatorhuis. All rights reserved.
 *
 */
#include <boost/algorithm/string.hpp>
#include <vector>
#include "dragon3d/rib.h"

std::istream& read_rib(std::istream& in, std::vector<RibNode>& ribs) {
	if(in) {
		std::string line;
		getline(in, line);
		boost::trim(line);
		if (line.size() > 0) {
			RibNode rn;
			rn.content = line;
			ribs.push_back(rn);
		}
	}
	return in;
}

File: src/net/transformatorhuis/cgi/conversion/Rib.java
//
//  Rib.java
//  RendermanXML
//
//  Created by Olivier Van Acker on Wed Dec 24 2003.
//  Copyright (c) 2003 __MyCompanyName__. All rights reserved.
//

public abstract class Rib {

    private String riName;
    
    public Rib(String ribElement, String riName) {
        this.riName = riName;
    }
    
    public String getElement() {
        return "<" + riName + setAttributes() + ">" + setBody() + "</" + riName + ">";
    }

    private String setAttributes() {

    }
    
}
